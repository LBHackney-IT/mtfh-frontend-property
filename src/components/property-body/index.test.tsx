import React from "react";

import { render, server } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";
import { rest } from "msw";

import { PropertyBody } from ".";

import { Asset } from "@mtfh/common/lib/api/asset/v1";
import * as auth from "@mtfh/common/lib/auth/auth";

const assetData: Asset = {
  id: "769894bd-b0bc-47eb-a780-322372c2448f",
  assetId: "0019062023",
  assetType: "Block",
  assetLocation: {
    parentAssets: [],
    floorNo: "0",
    totalBlockFloors: 3,
  },
  assetAddress: {
    uprn: "10008355547",
    addressLine1: "123 Test Block(CAN BE DELETED)",
    addressLine2: "",
    addressLine3: "",
    addressLine4: "",
    postPreamble: "",
    postCode: "FK8 1FH",
  },
  assetManagement: {
    isCouncilProperty: true,
    managingOrganisation: "London Borough of Hackney",
    managingOrganisationId: "c01e3146-e630-c2cd-e709-18ef57bf3724",
    isTMOManaged: false,
    agent: "Agent Smith",
    areaOfficeName: "Test",
    owner: "Test",
  },
  assetCharacteristics: {
    numberOfBedrooms: 0,
    numberOfLifts: 0,
    numberOfLivingRooms: 0,
    windowType: "Wood",
    yearConstructed: "1991",
    numberOfSingleBeds: null,
    numberOfDoubleBeds: null,
    numberOfFloors: null,
    totalBlockFloors: null,
    heating: null,
    propertyFactor: null,
    architecturalType: null,
  },
  versionNumber: 0,
  tenure: {
    id: "c01e3146-e630-c2cd-e709-18ef57bf3724",
    paymentReference: "",
    type: "",
    startOfTenureDate: "",
    endOfTenureDate: "",
    isActive: false,
  },
  rootAsset: "",
  parentAssetIds: "",
  rentGroup: null,
};

beforeEach(() => {
  jest.resetAllMocks();

  jest.spyOn(auth, "isAuthorisedForGroups").mockReturnValue(true);

  server.use(
    rest.get(
      `/api/v1/cautionary-alerts/properties-new/${assetData.assetId}`,
      (req, res, ctx) => res(ctx.status(200), ctx.json({ alerts: [] })),
    ),
  );
});

test("it shows the new process button", () => {
  // Arrange
  const { container } = render(
    <PropertyBody assetDetails={assetData} childAssets={[]} enableNewProcesses />,
    {
      url: `/property/${assetData.id}`,
      path: "/property/:assetId",
    },
  );

  // Assert
  const newProcessButton = screen.getByText("New Process");

  expect(newProcessButton).toBeVisible();

  expect(container).toMatchSnapshot();
});

test("it hides the new process button", () => {
  // Arrange
  const { container } = render(
    <PropertyBody assetDetails={assetData} childAssets={[]} enableNewProcesses={false} />,
    {
      url: `/property/${assetData.id}`,
      path: "/property/:assetId",
    },
  );

  // Assert
  const newProcessButton = screen.queryByText("New Process");

  expect(newProcessButton).not.toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
