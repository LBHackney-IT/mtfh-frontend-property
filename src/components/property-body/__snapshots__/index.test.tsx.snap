// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it hides the new process button 1`] = `
<div>
  <div
    id="property-body-grid-container"
  >
    <div
      id="property-tree-grid-area"
    >
      <div>
        <div
          class="rst__tree"
        >
          <div
            class="rst__node"
            style="height: 62px;"
          >
            <div
              class="rst__lineBlock rst__lineHalfHorizontalRight"
              style="width: 44px;"
            />
            <div
              class="rst__nodeContent"
              style="left: 44px;"
            >
              <div
                style="height: 100%;"
              >
                <div>
                  <button
                    aria-label="Collapse"
                    class="rst__collapseButton"
                    style="left: -22px;"
                    type="button"
                  />
                  <div
                    class="rst__lineChildren"
                    style="width: 44px;"
                  />
                </div>
                <div
                  class="rst__rowWrapper"
                >
                  <div
                    class="rst__row"
                    style="opacity: 1;"
                  >
                    <div
                      class="rst__moveHandle"
                      draggable="true"
                    />
                    <div
                      class="rst__rowContents"
                    >
                      <div
                        class="rst__rowLabel"
                      >
                        <span
                          class="rst__rowTitle"
                        >
                          <span>
                            Hackney
                          </span>
                        </span>
                      </div>
                      <div
                        class="rst__rowToolbar"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rst__node"
            style="height: 62px;"
          >
            <div
              class="rst__lineBlock"
              style="width: 44px;"
            />
            <div
              class="rst__lineBlock rst__lineHalfVerticalTop rst__lineHalfHorizontalRight"
              style="width: 44px;"
            />
            <div
              class="rst__nodeContent"
              style="left: 88px;"
            >
              <div
                style="height: 100%;"
              >
                <div
                  class="rst__rowWrapper"
                >
                  <div
                    class="rst__row"
                    style="opacity: 1;"
                  >
                    <div
                      class="rst__moveHandle"
                      draggable="true"
                    />
                    <div
                      class="rst__rowContents"
                    >
                      <div
                        class="rst__rowLabel"
                      >
                        <span
                          class="rst__rowTitle"
                        >
                          123 Test Block(CAN BE DELETED) (this asset)
                        </span>
                      </div>
                      <div
                        class="rst__rowToolbar"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      id="repairs-grid-area"
    />
    <div
      id="comments-grid-area"
    >
      <h2
        class="lbh-heading-h2"
      >
        Comments
      </h2>
      <a
        class="govuk-button lbh-button"
        href="/comment/property/769894bd-b0bc-47eb-a780-322372c2448f"
      >
        Add comment
      </a>
      <div>
        <div
          class="mtfh-center mtfh-center--horizontal mtfh-center--vertical"
        >
          <svg
            class="mtfh-icon"
            focusable="false"
            height="50"
            stroke="#00703c"
            viewBox="0 0 42 42"
            width="50"
          >
            <title>
              Loading...
            </title>
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                stroke-width="5"
                transform="translate(3 3)"
              >
                <circle
                  cx="18"
                  cy="18"
                  r="18"
                  stroke-opacity=".5"
                />
                <path
                  d="M36 18c0-9.94-8.06-18-18-18"
                  transform="rotate(112.708 18 18)"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </path>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it shows the new process button 1`] = `
<div>
  <div
    id="property-body-grid-container"
  >
    <div
      id="property-tree-grid-area"
    >
      <div>
        <div
          class="rst__tree"
        >
          <div
            class="rst__node"
            style="height: 62px;"
          >
            <div
              class="rst__lineBlock rst__lineHalfHorizontalRight"
              style="width: 44px;"
            />
            <div
              class="rst__nodeContent"
              style="left: 44px;"
            >
              <div
                style="height: 100%;"
              >
                <div>
                  <button
                    aria-label="Collapse"
                    class="rst__collapseButton"
                    style="left: -22px;"
                    type="button"
                  />
                  <div
                    class="rst__lineChildren"
                    style="width: 44px;"
                  />
                </div>
                <div
                  class="rst__rowWrapper"
                >
                  <div
                    class="rst__row"
                    style="opacity: 1;"
                  >
                    <div
                      class="rst__moveHandle"
                      draggable="true"
                    />
                    <div
                      class="rst__rowContents"
                    >
                      <div
                        class="rst__rowLabel"
                      >
                        <span
                          class="rst__rowTitle"
                        >
                          <span>
                            Hackney
                          </span>
                        </span>
                      </div>
                      <div
                        class="rst__rowToolbar"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rst__node"
            style="height: 62px;"
          >
            <div
              class="rst__lineBlock"
              style="width: 44px;"
            />
            <div
              class="rst__lineBlock rst__lineHalfVerticalTop rst__lineHalfHorizontalRight"
              style="width: 44px;"
            />
            <div
              class="rst__nodeContent"
              style="left: 88px;"
            >
              <div
                style="height: 100%;"
              >
                <div
                  class="rst__rowWrapper"
                >
                  <div
                    class="rst__row"
                    style="opacity: 1;"
                  >
                    <div
                      class="rst__moveHandle"
                      draggable="true"
                    />
                    <div
                      class="rst__rowContents"
                    >
                      <div
                        class="rst__rowLabel"
                      >
                        <span
                          class="rst__rowTitle"
                        >
                          123 Test Block(CAN BE DELETED) (this asset)
                        </span>
                      </div>
                      <div
                        class="rst__rowToolbar"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      id="new-process-grid-area"
    >
      <a
        class="govuk-button lbh-button"
        href="/processes/property/769894bd-b0bc-47eb-a780-322372c2448f"
      >
        New Process
      </a>
    </div>
    <div
      id="repairs-grid-area"
    />
    <div
      id="comments-grid-area"
    >
      <h2
        class="lbh-heading-h2"
      >
        Comments
      </h2>
      <a
        class="govuk-button lbh-button"
        href="/comment/property/769894bd-b0bc-47eb-a780-322372c2448f"
      >
        Add comment
      </a>
      <div>
        <div
          class="mtfh-center mtfh-center--horizontal mtfh-center--vertical"
        >
          <svg
            class="mtfh-icon"
            focusable="false"
            height="50"
            stroke="#00703c"
            viewBox="0 0 42 42"
            width="50"
          >
            <title>
              Loading...
            </title>
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                stroke-width="5"
                transform="translate(3 3)"
              >
                <circle
                  cx="18"
                  cy="18"
                  r="18"
                  stroke-opacity=".5"
                />
                <path
                  d="M36 18c0-9.94-8.06-18-18-18"
                  transform="rotate(112.708 18 18)"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </path>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
